YvesIraguha/lost-and-found-api
# #171913621 setup development environment

#### What does this PR do?
Setup development environment 
#### Description of Task to be completed?
* Set up mongoose
* set up express
* set up mongodb
`https://www.digitalocean.com/community/tutorials/how-to-integrate-mongodb-with-your-node-application`
#### How should this be manually tested?
* clone the repo and checkout the branch `ch-setup-development-enviroment-#171913621`
* After checking out the branch, then,
* Copy the content of `.env` file and paste it in your `.env` file.
```javascript
    //.env file   copy data belew and edit where it is necessary for your .env file
USER = yourDbUsername
PASS = yourDbPassword
HOST = 127.0.0.1
PORT = 27017
DB_NAME = yourDbName
```
* Create node environment by running 
```
$export NODE_ENV="development" 
```
* Start the server by running `npm run dev`
* In the command prompt you should see **port** our app is running on and **Database name** it is connected on

*in Postman* 
> On `localhost:3000/lostAndFound/lost` send a *Post* request of simple data like
```json
{
    "name":"Habineza",
    "number":"0789121324
}
```
> You should see a successful message or unsuccessful message
#### What are the relevant pivotal tracker stories?
#171913621
